### PR TITLE
Implement grip threshold profile setting

### PR DESCRIPTION
**Pull request by FejZa**
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

This PR addresses https://github.com/XRTK/XRTK-Core/issues/665 and introduces a hand profile setting to adjust the hand grip threshold, if needed. Some users were running into issues that their grip would release too soon due to tracking imprecision. This new setting allows to lower the required grip / curl threshold of the hand for it to be considered gripping.

image

## Changes

- Fix: https://github.com/XRTK/XRTK-Core/issues/665
- Fix some inconsitencies with bold labels in profile inspectors
- Lowers default threshold

## Breaking Changes

This is a breaking change and bumps Core up to 0.2.2.

## Submodule Changes

- [SDK](https://github.com/XRTK/SDK/pull/222)
- [Lumin](https://github.com/XRTK/Lumin/pull/98)
- [Oculus](https://github.com/XRTK/Oculus/pull/104)
- [WMR](https://github.com/XRTK/WindowsMixedReality/pull/117)
- [Examples](https://github.com/XRTK/Examples/pull/27)

### CODE REVIEW

**Comment by FejZa** [comment]
Ok this one is now passing checks and ready for review.

File: XRTK-Core/Packages/com.xrtk.core/Runtime/Definitions/Controllers/Hands/BaseHandControllerDataProviderProfile.cs [hunk 1]
```diff
@@ -13,6 +13,16 @@ namespace XRTK.Definitions.Controllers.Hands
     /// </summary>
     public abstract class BaseHandControllerDataProviderProfile : BaseMixedRealityControllerDataProviderProfile
     {
+        [SerializeField]
+        [Range(.1f, 1f)]
+        [Tooltip("Threshold in range [0, 1] that defines when a hand is considered to be grabing.")]
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Runtime/Definitions/Controllers/Hands/BaseHandControllerDataProviderProfile.cs:18]: Tooltip says 0-1 bit the range attribute days .1-1

File: XRTK-Core/Packages/com.xrtk.core/Runtime/Definitions/Controllers/Simulation/Hands/SimulatedHandControllerDataProviderProfile.cs [hunk 2]
```diff
@@ -20,6 +20,16 @@ public class SimulatedHandControllerDataProviderProfile : SimulatedControllerDat
         /// </summary>
         public float HandPoseAnimationSpeed => handPoseAnimationSpeed;
 
+        [SerializeField]
+        [Range(.1f, 1f)]
+        [Tooltip("Threshold in range [0, 1] that defines when a hand is considered to be grabing.")]
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Runtime/Definitions/Controllers/Simulation/Hands/SimulatedHandControllerDataProviderProfile.cs:25]: She thing here

File: XRTK-Core/Packages/com.xrtk.core/Runtime/Definitions/InputSystem/MixedRealityInputSystemProfile.cs [hunk 3]
```diff
@@ -99,6 +99,16 @@ public SystemType GazeProviderType
 
         #region Global Hand Options
 
+        [SerializeField]
+        [Range(.1f, 1f)]
+        [Tooltip("Threshold in range [0, 1] that defines when a hand is considered to be grabing.")]
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Runtime/Definitions/InputSystem/MixedRealityInputSystemProfile.cs:104]: And again.

Kinda concerned about the amount of duplicate code. But we can address later

File: XRTK-Core/Packages/com.xrtk.core/package.json [hunk 4]
```diff
@@ -10,7 +10,7 @@
     "Mixed Reality",
     "DI"
   ],
-  "version": "0.2.2",
+  "version": "0.2.3",
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/package.json:13]: Might need to update the lock file too. It's still on 0.2.2

**Review by StephenHodgson — COMMENTED** [review]
Looks good just needs to address some issues with the tooltips and package versioning

**Review by StephenHodgson — DISMISSED** [review]

**Review by StephenHodgson — APPROVED** [review]

**Pull request merged**